ffi/capi/src/units_converter.rs

```diff
@@ -0,0 +1,147 @@
+pub mod ffi {
+    use crate::{errors::ffi::ICU4XError, provider::ffi::ICU4XDataProvider};
+    use alloc::boxed::Box;
+    use icu_experimental::units::converter::UnitsConverter;
+    use icu_experimental::units::converter_factory::ConverterFactory;
+    use icu_experimental::units::measureunit::MeasureUnit;
+    use icu_experimental::units::measureunit::MeasureUnitParser;
+
+    #[diplomat::opaque]
+    /// An ICU4X Units Converter Factory object, capable of creating converters a [`ICU4XUnitsConverter`]
+    /// for converting between two [`ICU4XMeasureUnit`]s.
+    /// Also, it can parse the CLDR unit identifier (e.g. `meter-per-square-second`) and get the [`ICU4XMeasureUnit`].
+    #[diplomat::rust_link(icu::experimental::units::converter_factory::ConverterFactory, Struct)]
+    #[diplomat::rust_link(
+        icu::experimental::units::converter_factory::ConverterFactory,
+        Struct,
+        hidden
+    )]
+    pub struct ICU4XUnitsConverterFactory(pub ConverterFactory);
+
+    impl ICU4XUnitsConverterFactory {
+        /// Construct a new [`ICU4XUnitsConverterFactory`] instance.
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::new,
+            FnInStruct
+        )]
+        #[diplomat::attr(all(supports = constructors, supports = fallible_constructors), constructor)]
+        pub fn create(
+            provider: &ICU4XDataProvider,
+        ) -> Result<Box<ICU4XUnitsConverterFactory>, ICU4XError> {
+            Ok(Box::new(ICU4XUnitsConverterFactory(call_constructor!(
+                ConverterFactory::new [r => Ok(r)],
+                ConverterFactory::try_new_with_any_provider,
+                ConverterFactory::try_new_with_buffer_provider,
+                provider,
+            )?)))
+        }
+
+        /// Creates a new [`ICU4XUnitsConverter`] from the input and output [`ICU4XMeasureUnit`]s.
+        /// Returns nothing if the conversion between the two units is not possible.
+        /// For example, conversion between `meter` and `second` is not possible.
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::converter,
+            FnInStruct
+        )]
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::converter,
+            FnInStruct,
+            hidden
+        )]
+        pub fn converter(
+            &self,
+            from: &ICU4XMeasureUnit,
+            to: &ICU4XMeasureUnit,
+        ) -> Option<Box<ICU4XUnitsConverter>> {
+            Some(ICU4XUnitsConverter(self.0.converter(&from.0, &to.0)?).into())
+        }
+
+        /// Creates a parser to parse the CLDR unit identifier (e.g. `meter-per-square-second`) and get the [`ICU4XMeasureUnit`].
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::parser,
+            FnInStruct
+        )]
+        #[diplomat::rust_link(
```

Comment:
@Manishearth , is this `rust_link`  correct? or should be getter? but the parser is not a struct field?

Reply:
Again, you have duplicate rust_links

Reply:
done